### PR TITLE
Add deep merge mixin to cashbox model

### PR DESCRIPTION
**Pull request by maulin**

### CODE REVIEW

File: lib/cashbox/model/model.rb [hunk 1]
```diff
@@ -5,5 +5,6 @@ class Model < Hashie::Trash
     include Hashie::Extensions::IndifferentAccess
     include Hashie::Extensions::Dash::PropertyTranslation
     include Hashie::Extensions::Dash::Coercion
+    include Hashie::Extensions::DeepMerge
```
Comment (dannyburrows) [inline lib/cashbox/model/model.rb]: What is the use case for adding the DeepMerge?

Comment (maulin) [reply lib/cashbox/model/model.rb]: without the mixin, it throws errors when you try to do a deep merge but there are undefined (extra) params that are not properties on the model

File: spec/unit/model/model_spec.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require 'spec_helper'
+
+describe Cashbox::Model do
+  it { is_expected.to be_a(Hashie::Trash) }
```
[the remark this answers is not in the document]
Comment (maulin) [reply spec/unit/model/model_spec.rb]: theres even a hashie mash :)

File: lib/cashbox/model/model.rb (hunk 1 above)
Comment (jonstorer) [reply lib/cashbox/model/model.rb]: what happens to the fields that aren't defined? do we magically get them?

Comment (maulin) [reply lib/cashbox/model/model.rb]: nope. throws a nomethod error

**Pull request closed without merging**